sdk/avalon_sdk/ethereum/ethereum_listener.py

```diff
@@ -66,12 +66,20 @@ def __init__(self, config):
             callback(event, *kargs, **kwargs)
             self.queue.task_done()
 
-    async def sync_handler(self, *kargs, **kwargs):
+    async def sync_handler(self, check_event_callback=None, *kargs, **kwargs):
         logging.info("Started synchronous handler to handle an event")
-        event = await self.queue.get()
-        logging.debug("Event popped from listener queue")
+        while True:
+
+            event = await self.queue.get()
+            logging.debug("Event popped from listener queue")
+            self.queue.task_done()
+
+            # Break out of the loop if check_event_callback method is
+            # defined and returns True when called.
+            if check_event_callback is not None and \
+                    check_event_callback(event, *kargs, **kwargs) is True:
+                break
```

Comment:
If check_event_callback  is None is this an error  case? If yes, return an error. If not, please explain why.